testsuite/tests/contrib/lkt_semantic/array_type_concrete/test.lkt

```diff
@@ -0,0 +1,15 @@
+val i: Int = 1
+# TODO: should infer type from i
+val a = [i, 2, 3]
+
+val s: String = "str1"
+# TODO: should infer type from s
+val a = [str, "str2"]
+
+fun foo(): Int = 1
+# TODO: should infer type from foo()
+val a = [foo(), 2, 3]
+
+fun bar(): String = "str1"
+# TODO: should infer type from bar()
+val a = [foo(), "str2", "str3"]
```

Comment:
Pasto: should be `bar()`

Also, I think it would be valuable to include an invalid case that mixes type inference and inconsistent types. For instance:

```
val a4 = [foo(), bar()]
```

Reply:
Will be done